### PR TITLE
Chore/new docs update contributor guides

### PR DESCRIPTION
**Pull request by saltcod**
This can only be merged AFTER we've launched the new docs.

### CODE REVIEW

File: .github/workflows/docs-compilation-check.yml [hunk 1]
```diff
@@ -35,8 +35,8 @@ jobs:
 
       - name: Install Deps
         run: npm ci
-        working-directory: ./apps/reference
+        working-directory: ./apps/docs
 
       - name: Build
         run: npm run build
```
Comment (joshenlim) [inline .github/workflows/docs-compilation-check.yml:41]: we may not need this file anymore actually as it'll build on vercel
but if we want to keep this - need to change the build command to use turbo

File: DEVELOPERS.md [hunk 2]
```diff
@@ -183,7 +183,7 @@ Then edit and visit any of the following sites:
 Site | Directory | Description | Local development server
 ---- | --------- | ----------- | ------------------------
 [supabase.com](https://supabase.com) | `/apps/www` | The main website | http://localhost:3000
-[supabase.com/docs](https://supabase.com/docs) | `apps/reference` | Guides and Reference documentation | http://localhost:3010/docs
+[supabase.com/docs](https://supabase.com/docs) | `apps/docs` | Guides and Reference documentation | http://localhost:3010/docs
```
Comment (joshenlim) [inline DEVELOPERS.md:186]: localhost3001

File: apps/reference/README.md [hunk 3]
```diff
@@ -42,4 +42,4 @@ We use documentation specifications which can be used to generate human-readable
 - CLISpec (custom to Supabase): for CLI commands and usage.
 
 The benefit of using custom specifications is that we can generate many other types from a strict schema (eg, HTML and manpages).
-It also means that we can switch any documentation system we want. On this site we use Docusaurus, but in Supabase's official website we use a custom React site and expose only a subset of the available API for each tool.
+It also means that we can switch any documentation system we want. On this site we use Next.JS, but in Supabase's official website we use a custom React site and expose only a subset of the available API for each tool.
```
Comment (joshenlim) [inline apps/reference/README.md:45]: just to sanity check - this is in the `reference` folder though?
we probably need the same README in `temp-docs`

Comment (saltcod) [reply apps/reference/README.md:45]: right. Copied this file in the new docs readme

**Review by joshenlim — APPROVED** [review]

**Pull request merged**